lib/foreman_bootdisk/engine.rb

```diff
@@ -77,6 +77,8 @@ class Engine < ::Rails::Engine
     config.to_prepare do
       begin
         Host::Managed.send(:prepend, ForemanBootdisk::HostExt)
+        Host::Managed.send(:extend, ForemanBootdisk::RendererMethods)
+        Host::Managed.send(:include, ForemanBootdisk::RendererMethods)
```

Comment:
Shouldn't this be covered by [this](https://github.com/theforeman/foreman/blob/91c553d00ad497df40ac5bdd28945b5f29f34020/app/models/host/managed.rb#L168) and [this](https://github.com/theforeman/foreman_bootdisk/blob/master/lib/foreman_bootdisk/engine.rb#L67)?

Reply:
This is actually the fix. The method was made visible only in controller context (`extend_template_helpers`), not in host context. There is no plugin API for that, it is perhaps assumed that plugins include their own concern into Host class.

This needs overhaus as I said, extending Host class like that is messy, Host is God-object in the current design. But that's out of table for now :)

Reply:
Actually, Host is the `Oh, my god!` object. :-D